### PR TITLE
softmax translation

### PR DESCRIPTION
**Pull request by varunjain99**
translation for softmax
@bddppq @yinghai @Maratyszcza

### CODE REVIEW

**Comment by varunjain99** [comment]
@yinghai I was waiting for my unit test PR to go through before adding, though I was applying those unit tests locally. Now that it has gone through I have added the tests here :)

File: onnx_xla/operator_registry.cc [hunk 1]
```diff
@@ -34,4 +34,20 @@ TranslationMap& OperatorRegistry::map() {
   static TranslationMap map;
   return map;
 }
+
+XlaComputation OperatorRegistry::add(PrimitiveType dataType) {
+  XlaBuilder builder("add");
+  auto y = builder.Parameter(0, ShapeUtil::MakeShape(dataType, {}), "y");
+  auto x = builder.Parameter(1, ShapeUtil::MakeShape(dataType, {}), "x");
+  builder.Add(y, x);
+  return builder.Build().ConsumeValueOrDie();
```
Comment (yinghai) [inline onnx_xla/operator_registry.cc]: Noob question: what does `ConsumeValueOrDie` do?

Comment (varunjain99) [reply onnx_xla/operator_registry.cc]: It takes a StatusOr object, checks the status is ok, and returns the object itself

Comment (yinghai) [reply onnx_xla/operator_registry.cc]: Hmm, what does the `Die` part do?

Comment (varunjain99) [reply onnx_xla/operator_registry.cc]: Tensorflow throws its error explaining why the computation couldn't be built. As an example, I intentionally screwed up broadcast dimensions in the sum operator translation and got this message `Invalid argument: Broadcast dimensions field must either be not set or be the identity on binary operations with operands of the same rank. Aborted`

Comment (yinghai) [reply onnx_xla/operator_registry.cc]: No, it's OK. I was just curious about its behavior.
